### ISSUE TITLE
Make Geocoding easier (aka make massive geocoding possible and easy in CitySDK)

### ISSUE THREAD

**Issue by tyronegrandison** [uscensusbureau/citysdk#85]
Got confirmation for the 4th time in a couple of days that the first step (and the most difficult) is geocoding the locations of the entities that one is analyzing.

The next steps are then placing these geocoded entities in Census regions, adding Census data on top of that region and performing analysis. 

Unfortunately just adding the CENSUS geocoder is not sufficient, the CENSUS geocoder needs to be improved. 

CitySDK should provide geocoding support now and lobby for internal improvements.

**Comment by afrasier-bah** [comment]
Could you give some examples of what users would like to do?

**Comment by tyronegrandison** [comment]
1) I want to analyze accident and demographic data to influence policy and resource allocation decisions to improve the number of fatalities in DC. I have the addresses of accidents from local Department of Transportation. I can get demographics from Census. My first step is geocoding the accident addresses. 

2) I want to see the impact that a federal agency is having on a population group. I have the locations of the offices of the agency. I have population stats from Census. Same issue. 

Whenever you are looking to combine a specific data of interest with Census information, you always have to start with geocoding the entities in the data set.

**Comment by afrasier-bah** [comment]
Yes. We could add support in the SDK for Geocoding the addresses (i.e. 123 Green Street) rather than just Lat/Lng as we currently have. Then you could grab the census data via address without first having to grab lat/lng. Would that help the problem or did I misinterpret the issue?

**Comment by tyronegrandison** [comment]
That would be a great (and useful) first step.

The downstream problem then becomes on do you visualize your work. In the current way of doing this means that all the data sets must map to lat/long.

So, no matter how you slice it, also having the ability to geocode addresses in CitySDK would be useful. I am also guessing that it would be a simple (yet catalytic) function that would drive rapid adoption (because doing geocoding is so painful today).

**Comment by tyronegrandison** [comment]
A quick wrapper around the CENSUS GeoCoder would be great. Something like this https://github.com/SigmaMonstR/census-geocoder

In addition to address-to-census-data functionality

**Comment by afrasier-bah** [comment]
ZIP code functionality introduced in e9d4feafb6e9ce2645e7f1ed5f2abc828d3468da
Street address functionality to come in the future.

**Comment by afrasier** [comment]
(Coming in the next push from the Census Internal Servers) 

Users can now specify location via an address object:

```
var request = {
      address:  {
           street: "18 F Street NW",
           city: "Washington",
           state: "DC"
      }
}
```

Which then extends our location specification options to: address, zip code, state only, or lat/lng. The geocoder doesn't support ONLY City, State; for some reason you must have a street. I'll see what I can do about this later on.

**Comment by afrasier** [comment]
Resolved f1cabdf45be7797e9b3266c39a6b0db60db7549e
